src/twisted/enterprise/adbapi.py

```diff
@@ -286,8 +286,7 @@ def runWithConnection(self, func, *args, **kw):
             C{func(Transaction(...), *args, **kw)}, or a
             L{twisted.python.failure.Failure}.
         """
-        from twisted.internet import reactor
-        return threads.deferToThreadPool(reactor, self.threadpool,
+        return threads.deferToThreadPool(self._reactor, self.threadpool,
```

Comment:
It'd be good to have a test that passes a stub `IReactorThreads` provider and asserts the thread pool uses it.